utils/query/createGetManyQuery.ts

```diff
@@ -3,29 +3,32 @@ import { z } from 'zod';
 import { OptionalKeys } from 'types';
 import { generateFilterForModelSchema } from './filter';
 
+type OptionalKeysNoCreatedAtUpdatedAt<T> = Exclude<OptionalKeys<T>, 'createdAt' | 'updatedAt'>;
+
 export function nonRelationFields(schema: z.ZodObject<any>) {
 	return Object.entries(schema.shape)
 		.map(([key, value]) => {
+			if (key === 'createdAt' || key === 'updatedAt') {
+				return key;
+			}
```

Comment:
to create the `filter` query param schema, i filter out all `optional()` fields, which are normally associations and I don't want to allow filtering by association (e.g. show me all collections with a Pub with name 'My Pub', too complex).
However, due to annoying reasons, I needed to make the `createdAt` and `updatedAt` optional as well, so I allow an exception for them.